### PR TITLE
Port to platformio

### PR DESCRIPTION
**Pull request by max-at-groundlight**
The `platformio.ini` file describes the board and the libraries, so deploying is quite trivial with the Platform IO extension. Readme is updated to be a little more verbose. Arduino IDE support is still present.

### CODE REVIEW

File: edgelight/include/README
Comment (positavi) [inline edgelight/include/README:1]: i don't think we need a readme for this.

File: edgelight/lib/README
Comment (positavi) [inline edgelight/lib/README:1]: i don't think we need a readme here

File: edgelight/src/.DS_Store
Comment (positavi) [inline edgelight/src/.DS_Store:1]: this shouldn't be in here.  its a private file from your laptop.

File: edgelight/.gitignore
Comment (positavi) [inline edgelight/.gitignore:1]: i don't think this is needed especially for hidden files.

File: edgelight/.vscode/extensions.json
Comment (positavi) [inline edgelight/.vscode/extensions.json:1]: not needed.

**Review by positavi — CHANGES_REQUESTED** [review]
there are a bunch of files included here inadvertently that shouldn't be here, e.g. the .DS_Store files from your laptop.  

probably don't need the .gitignore for this or the .vscode.  you can update the main readme with the link the platformio install docs. 

i don't think you need readme files in all the subdirectories.  maybe those are default from platformIO setup ?  ideally eliminate the directories if there are no files in there anyway.   you could try eliminating them from the repo and seeing if platformio automatically creates them locally when opening the project, which is probably ideal. 

i would expect the only files in this repo are :

README.md
platformio.ini
edgelight.ino

**Comment by max-at-groundlight** [comment]
@positavi

.vscode is automatically generated but the test, lib, and include directories are not. You are correct that those directories and their README files are generated when creating a new platform io project, but not when opening an existing one.

Also I believe that we should keep the generated .gitignore because it keeps git from including the .pio folder containing all of the build files and downloaded libraries.

Would you still like me to remove the directories and/or the .gitignore file?

**Review by positavi — CHANGES_REQUESTED** [review]
looks like there is still a .DS_Store file in the src directory to be removed.

agree to keep the .gitignore to keep from accidentally adding piles of files down the road.

maybe just trim the README to something very short saying this directory is part of the platformIO structure but not needed for this project .. i.e. justifying that it is otherwise empty.

**Pull request closed without merging**